### PR TITLE
Use real Daisuki database for tests

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/test/ci/song_selector.test.ts [hunk 1]
```diff
@@ -60,380 +47,380 @@ describe("getFilteredSongList", () => {
...
+            it("should only return the songs published in or after 2016", async () => {
                 await guildPreference.setBeginningCutoffYear(2016);
                 const { songs } = await SongSelector.getFilteredSongList(
                     guildPreference
                 );
 
-                assert.strictEqual(songs.size, expectedSongCount);
+                assert.strictEqual(
+                    Array.from(songs).every(
+                        (song) => song.publishDate > new Date("2016-01-01")
```
Comment (taahamahdi) [inline src/test/ci/song_selector.test.ts:220]:
```suggestion
                        (song) => song.publishDate >= new Date("2016-01-01")
```

File: src/test/ci/song_selector.test.ts [hunk 2]
```diff
@@ -60,380 +47,380 @@ describe("getFilteredSongList", () => {
...
-                        artistWithCollabingSubunit.id,
-                        subunitWithCollab.id,
-                        subunitCollabArtist.id,
-                        parentCollabArtist.id,
-                    ].includes(song.id_artist)
-                );
+    describe("include subunits (and the subunit has a collab)", () => {
+        it("should match the songs from the group, collabs of that group, and collabs of any subunits of that group", async () => {
+            const artistWithCollabingSubunit = { name: "BIGBANG", id: 28 };
+            // F
```
Comment (taahamahdi) [inline src/test/ci/song_selector.test.ts:340]: Can probably get rid of these comments

File: src/test/ci/song_selector.test.ts [hunk 3]
```diff
@@ -562,24 +547,20 @@ describe("getFilteredSongList", () => {
...
                 const filteredSongs = (
                     await SongSelector.getFilteredSongList(guildPreference)
                 ).songs;
 
-                const femaleOrCoedSongCount = [...filteredSongs].filter((x) =>
-                    [Gender.FEMALE, Gender.COED].includes(x.members)
-                ).length;
-
-                for (let i = 0; i < femaleOrCoedSongCount; i++) {
+                for (let i = 0; i < 10; i++) {
```
Comment (taahamahdi) [inline src/test/ci/song_selector.test.ts:563]: Can the for loop go over all female or coed songs?

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/test/test_setup.ts [hunk 4]
```diff
@@ -1,128 +1,31 @@
...
-}
-
 before(async function () {
-    this.timeout(10000);
+    if (process.env.NODE_ENV !== EnvType.TEST) {
+        logger.error("Must be running with NODE_ENV=EnvType.TEST");
+        process.exit(1);
+    }
+
+    this.timeout(20000);
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/test/test_setup.ts:27]: Yes, it takes extra time to import the database and some of the tests run a bit longer than real dataset

File: src/test/ci/song_selector.test.ts [hunk 5]
```diff
@@ -596,26 +577,17 @@ describe("selectRandomSong", () => {
...
                 const filteredSongs = (
                     await SongSelector.getFilteredSongList(guildPreference)
                 ).songs;
 
-                const maleOrCoedSongCount = [...filteredSongs].filter((x) =>
-                    [Gender.MALE, Gender.COED].includes(x.members)
-                ).length;
-
-                for (let i = 0; i < maleOrCoedSongCount; i++) {
+                for (let i = 0; i < 10; i++) {
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/test/ci/song_selector.test.ts:590]: It doesn't really need to, also it makes the test run too long

File: src/test/ci/song_selector.test.ts (hunk 3 above)
Comment (Brainicism) [reply src/test/ci/song_selector.test.ts:563]: See below

File: src/test/ci/game_utils.test.ts [hunk 6]
```diff
@@ -216,17 +229,18 @@ describe("song query", () => {
                     });
                 });
 
-                describe("names match excluding punctuation", () => {
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/test/ci/game_utils.test.ts:219]: Perhaps but this test wasn't even testing that

File: src/test/ci/song_selector.test.ts [hunk 7]
```diff
@@ -455,103 +442,101 @@ describe("getFilteredSongList", () => {
...
-                ).length;
-
+            it("should only return the korean songs", async () => {
                 await guildPreference.setLanguageType(LanguageType.KOREAN);
                 const { songs } = await SongSelector.getFilteredSongList(
                     guildPreference
                 );
 
-                assert.strictEqual(songs.size, expectedSongCount);
+                // every language tag has atleast one song
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/test/ci/song_selector.test.ts:455]: This isn't equivalent

[the remark this answers is not in the document]
Comment (taahamahdi) [reply src/test/ci/song_selector.test.ts:455]: I might be misinterpreting the original comment then. I thought you were making sure all songs have no foreign language tags with the intersection

[the remark this answers is not in the document]
Comment (Brainicism) [reply src/test/ci/song_selector.test.ts:455]: There's atleast one song of each language, ill reword

**Review by taahamahdi — APPROVED** [review]

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**